core/src/main/scala/org/apache/spark/rdd/ParallelCollectionRDD.scala

```diff
@@ -127,18 +127,15 @@ private object ParallelCollectionRDD {
       })
     }
     seq match {
-      case r: Range.Inclusive => {
-        val sign = if (r.step < 0) {
-          -1
-        } else {
-          1
-        }
-        slice(new Range(
-          r.start, r.end + sign, r.step).asInstanceOf[Seq[T]], numSlices)
-      }
       case r: Range => {
-        positions(r.length, numSlices).map({
-          case (start, end) =>
+        // 1 to Int.MaxValue and (-2 to Int.MinValue by -1) can trigger exclusive range int overflow
+        val needsInclusiveRange = r.isInclusive && (r.end == Int.MaxValue || r.end == Int.MinValue)
```

Comment:
this need not be specific to `Int.MaxValue` and `Int.MinValue`. We can just do a special case for inclusive range in general and I think the code will be significantly more straightforward to read. I'll explain below.